### ISSUE TITLE
[BUG] Cannot map property from String to Object even if it's marked as "ignore"

### ISSUE THREAD

**Issue by indalaterre** [mapstruct/mapstruct#2845]
Hi all I have the follow mapper configuration

```
@Mapping(source = "zip", target = "postalCode")
@Mapping(source = "type", target = "addressType")
@Mapping(source = "city", target= "city", ignore = true)
@Mapping(source = "order", target = "displaySequenceNumber")
```

In the left side of the mapper **city** is a String and in the right side is an object of class City
I want to ignore this mapping since I need to implement some custom logic fetching data from a DB that's why I decide to ignore it.

But I still have this error and viceversa
`Can't map property "String city" to "City city". Consider to declare/implement a mapping method: "City map(String value)"`

This is the workaround I implement that force me to think that this is a bug.

```
public City mapCityFromString(String city) {
	return null;
}

public String mapCityFromObject(City city) {
	return null;
}
```

As we know in the mapper implementation, these methods are never used because the fields are marked as ignored.

So to recap:

**Actual**
Can't map different type properties is fired even if the mapping of those properties is marked as "ignore"

**Expected**
The error is fired only if the mapping is requested

**Comment by filiphr** [comment]
I have tried this with 1.5.0.RC1 and it works as expected. Therefore, I am closing it.

If you can still reproduce it with 1.5.0.RC1 please provide a full example.
